src/Symfony/Component/Serializer/Normalizer/CacheableSupportsInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+/**
+ * Marker interface for normalizers and denormalizers that return
+ * an integer in their supports*() methods.
+ *
+ * By implementing this interface, the return value of the
+ * supports*() methods will be cached by type and format, based on the returned integer.
+ *
+ * @author Jeroen Spee <https://github.com/Jeroeny>
+ */
+interface CacheableSupportsInterface
```

Comment:
Couldn't we use the existing interface instead and deprecate returning a bool?

Reply:
Yes, that's also a possibility and makes adopting the new return type possibly easier. I've removed this interface.